backend/src/apiserver/resource/semaphore_manager.go

```diff
@@ -0,0 +1,106 @@
+package resource
+
+import (
+	"context"
+	"os"
+
+	"github.com/golang/glog"
+	"github.com/kubeflow/pipelines/backend/src/apiserver/client"
+	"github.com/kubeflow/pipelines/backend/src/common/util"
+	v1 "k8s.io/api/core/v1"
+	k8errors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Common constants for semaphore functionality
+const (
+	SemaphoreConfigName = "semaphore-config"
```

Comment:
Maybe prefix it with `ml-pipeline-` so that it's easily recognized as a KFP thing?